Microsoft.Toolkit.Uwp.UI.Controls/OrbitView/OrbitView.cs

```diff
@@ -0,0 +1,687 @@
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Microsoft.Toolkit.Uwp.UI.Animations;
+using Windows.UI;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Shapes;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// ItemsControl that lays out items in a circle with support for odbits and anchors
+    /// </summary>
+    [TemplatePart(Name = "AnchorCanvas", Type = typeof(Canvas))]
+    [TemplatePart(Name = "OrbitGrid", Type = typeof(Grid))]
+    [TemplatePart(Name = "CenterContent", Type = typeof(ContentPresenter))]
+    public sealed class OrbitView : ItemsControl
+    {
+        private const double _animationDuration = 200;
+
+        private OrbitViewPanel _panel;
+        private Grid _orbitsContainer;
+        private Canvas _anchorCanvas;
+        private ContentPresenter _centerContent;
+        private Compositor _compositor;
+
+        private Dictionary<object, Ellipse> _orbits;
+        private Dictionary<object, Line> _anchors;
+
+        /// <summary>
+        /// Raised when an item has been clicked or activated with keyboard/controller 
+        /// </summary>
+        public event EventHandler<OrbitViewItemClickedEventArgs> ItemInvoked;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrbitView"/> class.
+        /// Creates a new instance of <see cref="OrbitView"/>
+        /// </summary>
+        public OrbitView()
+        {
+            DefaultStyleKey = typeof(OrbitView);
+
+            IsTabStop = false;
+            TabNavigation = KeyboardNavigationMode.Once;
+            KeyDown += OrbitView_KeyDown;
+            _orbits = new Dictionary<object, Ellipse>();
+            _anchors = new Dictionary<object, Line>();
+
+            if (Windows.ApplicationModel.DesignMode.DesignModeEnabled)
+            {
+                var items = new List<OrbitViewItem>();
+                items.Add(new OrbitViewItem() { Distance = 0.1, Diameter = 0.5, Label = "test" });
+                items.Add(new OrbitViewItem() { Distance = 0.1, Diameter = 0.5, Label = "test" });
+                items.Add(new OrbitViewItem() { Distance = 0.1, Diameter = 0.5, Label = "test" });
+                ItemsSource = items;
+            }
+        }
+
+        protected override void OnApplyTemplate()
+        {
+            _centerContent = (ContentPresenter)GetTemplateChild("CenterContent");
```

Comment:
This should be as instead of casting because if CenterContent is not ContentPresenter or null it will throw an exception.